scipy/sparse/linalg/_special_sparse_arrays.py

```diff
@@ -511,3 +513,429 @@ def _adjoint(self):
+        a += np.diag(d1, 1) + np.diag(d1, -1)
+        a += np.diag(d2, 2) + np.diag(d2, -2)
+        return a
+    
+    def _matvec(self, x):
+        """
+        Construct matrix-free callable banded-matrix-vector multiplication by
+        the Sakurai matrix without constructing or storing the matrix itself
+        using the knowledge of its entries and the 5-diagonal format.
+        """
+        n = self.n
+        assert n == x.shape[0]
+        x = x.reshape(n, -1)
+        sx = np.zeros_like(x)
+        sx[0,:] = 5 * x[0,:] - 4 * x[1,:] + x[2,:]
+        sx[-1,:] = 5 * x[-1,:] - 4 * x[-2,:] + x[-3,:]
+        sx[1:-1,:] = (6 * x[1:-1,:] - 4 * (x[:-2,:] + x[2:,:])
+                      + np.pad(x[:-3,:], ((1,0),(0,0)))
+                      + np.pad(x[3:,:], ((0,1),(0,0))))
+        return sx
+
+    def _matmat(self, x):
+        """
+        Construct matrix-free callable matrix-matrix multiplication by
+        the Sakurai matrix without constructing or storing the matrix itself
+        by reusing the ``_matvec(x)`` that supports both 1D and 2D arrays ``x``.
+        """        
+        return self._matvec(x)
+
+    def _adjoint(self):
+        return self
+
+    def _transpose(self):
+        return self
+
+
+class MikotaM(LinearOperator):
+    """
+    Construct a mass matrix in various formats of Mikota pair.
+
+    The mass matrix `M` is square real diagonal
+    positive definite with entries that are reciprocal to integers.
+
+    Parameters
+    ----------
+    shape : tuple of int
+        The shape of the matrix.
+    dtype : dtype
+        Numerical type of the array. Default is ``np.float64``.
+
+    Methods
+    -------
+    toarray()
+        Construct a dense array from Mikota data
+    tosparse()
+        Construct a sparse array from Mikota data
+    tobanded()
+        The format for banded symmetric matrices,
+        i.e., (1, n) ndarray with the main diagonal.
+    """
+    def __init__(self, shape, dtype):
+        self.shape = shape
+        self.dtype = dtype
+        super().__init__(dtype, shape)
+
+    def tosparse(self):
+        from scipy.sparse import diags
+        n = self.shape[0]
+        aranp1 = np.arange(1, n + 1, dtype=self.dtype)
+        aranp1_inv = 1. / aranp1
+        return diags([aranp1_inv], [0], shape=(n, n))
+
+    def tobanded(self):
+        n = self.shape[0]
+        aranp1 = np.arange(1, n + 1, dtype=self.dtype)
+        return 1. / aranp1
+
+    def toarray(self):
+        n = self.shape[0]
+        aranp1 = np.arange(1, n + 1, dtype=self.dtype)
+        aranp1_inv = 1. / aranp1
+        return np.diag(aranp1_inv)
+
+    def _matvec(self, x):
+        """
+        Construct matrix-free callable banded-matrix-vector multiplication by
+        the Mikota mass matrix without constructing or storing the matrix itself
+        using the knowledge of its entries and the diagonal format.
+        """
+        n = self.shape[0]
+        aranp1_inv = 1. / np.arange(1, n + 1)
```

Comment:
Similar changes here; not 100% what would be best, but at least the following should work
```suggestion
        aranp1_inv = self._diag
```

Reply:
Yes, I forgot that matvec here uses the diagonal vector as well. I.e., this original suggestion in this class is the best to do. Testing it now.